tekore/client/base.py

```diff
@@ -121,25 +121,54 @@ def _request(
             self,
             method: str,
             url: str,
+            content_type: str = 'application/json',
             payload=None,
             params: dict = None
     ):
+        data = payload
+        if payload is not None and content_type == 'application/json':
+            data = json.dumps(payload)
+
         return Request(
             method=method,
             url=build_url(url),
-            headers=self._create_headers(),
+            headers=self._create_headers(content_type),
             params=parse_url_params(params),
-            data=json.dumps(payload) if payload is not None else None
+            data=data
```

Comment:
Is the JSON dump not required? That's neat if it's true.

Reply:
It is dumped above if `content-type` is JSON. I had to introduce it because of errors that I encountered when working on playlist image adding endpoint

Reply:
Oh, I missed it! Please see my review on the whole change.